portmap/core/articles.py

```diff
@@ -36,8 +39,18 @@ def get_content_files():
         logging.error(f"Error accessing Github API: {e}")
         raise RuntimeError("Failed to access Github API.")
 
-    return debug_articles_info
+    # Generate a static copy of the root index.html
+    try:
+        if not os.path.exists(settings.STATIC_ROOT):
```

Comment:
Did you discover if this is a regular thing - that we need to make STATIC_ROOT every time we have a fresh deploy, for example?  I'm curious about what the conditions are to need to create it.  If it's just to make local development work smoothly that's OK too but I'd love a comment here about that.

Reply:
`STATIC_ROOT` is the directory created by `manage.py collectstatic`, so it's just dependent on whether this runs before or after that command. Our current deploy workflow does run this code before `collectstatic`, so it will create the directory every time.

Thinking over this again, it might be better to put pre-generated HTML files in their own directory since `collectstatic` kind of expects to have control over `STATIC_ROOT`, and this will make a copy of the file accessible at portmap.dtinit.org/static/index.html. I'll still exclude the directory from git since the contents will be derivatives.

Reply:
Update: https://github.com/dtinit/portmap/pull/86/commits/d9076a03c2b330a94e85c143258e255325add798

I think this approach is more appropriate since `collectstatic` and what we're doing here are independent of one another.